### ISSUE TITLE
add logcosh loss

### ISSUE THREAD

**Issue by newhotter** [catboost/catboost#844]
For keras there is a logcosh loss func. For lgb there is a fair loss func. For xgboost there is a tweedie loss func. These three loss functions are better than the default MSE/RMSE loss function for many problems. 

https://www.kaggle.com/c/allstate-claims-severity/discussion/24520
Above is a disscussion for logcosh, fair and tweedie loss functions.

For fair loss function, it gives gradients and hess. I don't know how to implement them in catboost custom loss function.

**Comment by annaveronika** [comment]
There is a tutorial how to implement your loss function:
https://github.com/catboost/tutorials/blob/master/custom_loss/custom_metric_tutorial.md

But is a great idea to add the objectives that you requested to catboost.
I am labeling the issue as help wanted. If you will have time and will for thay, we will be greatful for a contribution!

**Comment by mdomarsaleem** [comment]
@annaveronika  I can give it a try if this is still an open issue.

**Comment by annaveronika** [comment]
It is an open issue, you are very welcome to try! Here was adding of some other metrics, which can be used as an example:
https://github.com/catboost/catboost/commit/50f98c5a93a7a26794596fe672a60b60e0a36a49

**Comment by annaveronika** [comment]
https://github.com/catboost/catboost/issues/577

**Comment by LyzhinIvan** [comment]
logcosh has been added in 9fd774716c4f246ca34bfc496d10eb30b4788f97
